runtime/ast/import.go

```diff
@@ -98,20 +97,25 @@ func LocationsMatch(first, second Location) bool {
 func LocationFromTypeID(typeID string) Location {
 	pieces := strings.Split(typeID, ".")
 
-	switch len(pieces) {
-	case 3:
-		if pieces[0] == AddressPrefix {
-			address, err := hex.DecodeString(pieces[1])
-			if err != nil {
-				return nil
-			}
+	if len(pieces) < 3 {
+		return nil
+	}
+
+	switch pieces[0] {
+	case IdentifierLocationPrefix:
+		return IdentifierLocation(pieces[1])
+
+	case StringLocationPrefix:
+		return StringLocation(pieces[1])
 
-			return AddressLocation(address)
+	case AddressLocationPrefix:
+		address, err := hex.DecodeString(pieces[1])
+		if err != nil {
+			return nil
```

Comment:
Do you think we could get rid of location prefixes? I know I added them in the first place, but I've realized they don't make much sense to the end-user and are just confusing overall.

Also, I feel like I'm often conflating two things when I think about locations:
- A place where Cadence code lives (let's say `origin`)
- An _identifier_ for a place where Cadence code lives (let's say `path`)

For example, say we are executing a transaction. Currently we pass in a `TransactionLocation` instance that just tells the runtime "this code originates from a transaction". In order words, the program's `origin` is the transaction. Because locations are also used for imports, we assume that every `origin` has a `path`, but this isn't really the case for transactions and scripts, which can't just be "imported".

What if we allowed origins to be anonymous? Something like this:

```go
type Origin interface {
  String() string // returns ID for the origin, but this isn't necessarily the import path
}

type Importer interface {
  Resolve(loc Location) (Program, Origin)
}

Reply:
The prefixes are used in the location *IDs*, which not meant for end-users, but for serialization: We need a way to differentiate where types are defined, and do so in a way that it can be serialized. The location IDs should never be shown to a user, if we do we should fix that.

Sorry, I'm not quite following with the origins/identifiers/paths, maybe we can chat about this in a call?

Reply:
Yep, sounds good! This isn't a blocker for this PR so we can talk about it separately.